src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(currentState, action.extraData);
+        result.push({ ...currentState });
+        break;
+      }
+
+      case 'removeProperties': {
+        action.keysToRemove.forEach((key) => {
+          if (key in currentState) {
```

Comment:
It is overkill here, if you delete property which is not one of the object's, it is just doing nothing, so it is ok